openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -48,113 +33,134 @@ def process(self, instance):
             )
             return
 
-        instance.data["writeNode"] = write_node
-        self.log.debug("checking instance: {}".format(instance))
+        # get colorspace and add to version data
+        colorspace = napi.get_colorspace_from_node(write_node)
 
-        # Determine defined file type
-        ext = write_node["file_type"].value()
+        if render_target == "frames":
+            self._set_existing_files_data(instance, colorspace)
 
-        # Get frame range
-        handle_start = instance.context.data["handleStart"]
-        handle_end = instance.context.data["handleEnd"]
+        elif render_target == "frames_farm":
+            collected_frames = self._set_existing_files_data(
+                instance, colorspace)
+
+            self._set_expected_files(instance, collected_frames)
+
+            self._add_farm_instance_data(instance)
+
+        elif render_target == "farm":
+            self._add_farm_instance_data(instance)
+
+        # set additional instance data
+        self._set_additional_instance_data(instance, render_target, colorspace)
+
+    def _set_existing_files_data(self, instance, colorspace):
+        """Set existing files data to instance data.
+
+        Args:
+            instance (pyblish.api.Instance): pyblish instance
+            colorspace (str): colorspace
+
+        Returns:
+            list: collected frames
+        """
+        collected_frames = self._get_collected_frames(instance)
+
+        representation = self._get_existing_frames_representation(
+            instance, collected_frames
+        )
+
+        # inject colorspace data
+        self.set_representation_colorspace(
```

Comment:
out of scope for this PR but the fact that this function exists within the `ColormanagedPyblishPluginMixin` doesn't add much benefit IMO as I have found myself requiring to call it like a staticmethod and there's nothing within the function that really requires the class attributes or anything, it could be simply a util module function.

Because of this reason, on this new `pyblish_functions` that you created in 3.16 (with the good intent to abstract some of the functions so they can be reused), you were forced to pass the mixin object as an arg https://github.com/ynput/OpenPype/blob/018677d34d8a74bbd849b129f567a11c8c3a8926/openpype/pipeline/farm/pyblish_functions.py#L446 to the function just so you can callt he method, you wouldn't need that at all and you could call directly `set_representation_colorspace` from wherever util module it lives on.

Also in a lot of these functions from `pyblish_functions` I think it would be more reusable to remove the need of having to pass the `pyblish.api.Instance` object around and breaking it down further to the parts required of the instance (i.e., `create_skeleton_instance(instance)` -> `create_skeleton_instance(instance_data, anatomy)`...). Creating the Instance object is not always obvious from outside the plugins scope and for the most parts you don't need the whole object to do whatever the function does. This might be me not knowing how to instantiate a pyblish.Instance but for the use case I had to use it for (on-demand remote publishes to Deadline from CLI) I could do exactly the same with just passing the instance data dictionary but I had to copy paste most of those functions and change its args

Reply:
I can see your points and thanks for your suggestions. Yeah you are right that we should rething some of those concepts. I would assume that best way to address those points will be to create two issues for enhancements. 

- Remove the function `set_representation_colorspace` from the `ColormanagedPyblishPluginMixin` and make it a util module function.
- Modify the **pyblish_functions** to remove the need for passing the **pyblish.api.Instance** object and break it down further into the required parts of the instance. For example, `create_skeleton_instance(instance)` should be modified to `create_skeleton_instance(instance_data, anatomy)`. 

I am resolving this comment as not related to the PR.